### DISCUSSION TITLE
I have a performance problem (draw frame spikes)

### DISCUSSION THREAD

**Discussion by Yuzuctus** [ppy/osu#26371] [Q&A]
My problem isn't a priority, but it's quite disturbing. 

For a few days now, I've been having stability problems with osu! Lazer in STD mode (I only play this one). 

I have this config: 
Ryzen 7 5700x, RX 6750 XT, 32gb ram 3600Mhz (In short, a pretty good PC). 

However, sometimes my maps freeze. In fact, I have the impression that the approach circles freeze and resume, so it's more like stuttering...

I mainly do 5* maps and it seems to happen mostly on fast maps.

I've tested several options 

Dx11, OpenGl, OpenGl Legacy. None of them solved the problem. 

In principle, I'm on "Basically unlimited", but I've tested the other modes ( x2 refresh, 4x refresh, 8x refresh) and sqa hasn't solved my problem. 
+I'm playing on a 165Hz screen

I tried clearing the cache but it didn't solve the problem. 

I applied the "High" priority to Osu! Lazer and Open Tablet and that didn't solve the problem either. 

I clicked on Compact and Block realm without knowing what it does to try to solve the problem but without success...

I wanted to share logs but couldn't find any... (I must not be very good)

**Comment by bdach** [comment]
read this and apply to it first https://osu.ppy.sh/wiki/en/Performance_troubleshooting

if nothing helps attach video of this with frame graph shown (ctrl-f11 twice)

> I applied the "High" priority to Osu! Lazer and Open Tablet and that didn't solve the problem either.

do NOT do this

> I clicked on Compact and Block realm without knowing what it does to try to solve the problem but without success...

do NOT do this it is 100% unrelated

**Reply by Yuzuctus** [reply]
To tell you the truth, I don't understand it anymore... 

I played a little for the stuttering but I had more trouble with an OBS chord... 

On the video I sent you, I felt a stutter on the lower left notes about 1 second into the video... 

However, on the bar there's only one peak, which could be the moment of the stutter, but if I look at the video I get the impression that everything's fine... 

https://github.com/ppy/osu/assets/93346083/28a63f37-a3aa-44f4-ac7c-10d060b8d631

**Reply by peppy** [reply]
There's definitely a draw spike on a regular interval. We'll need to figure out what's causing that.

**Reply by Yuzuctus** [reply]
I played again tonight and I had quite a bit of stuttering so I tried to take a clip that shows the graph better. I don't know if that helps.

https://github.com/ppy/osu/assets/93346083/b2680ee4-bf63-4eac-ae8c-83f1e808b3cf
